admin/client/src/state/breadcrumbs/BreadcrumbsActions.js

```diff
@@ -0,0 +1,40 @@
+import ACTION_TYPES from './BreadcrumbsActionTypes';
+
+/**
+ * Set selected changeset item
+ *
+ * @param {array} List of breadcrumbs to add, each of which is an object
+ * with a 'text' and optional 'href' property.
+ * @return {object}
+ */
+export function setBreadcrumbs(breadcrumbs) {
+  return {
+    type: ACTION_TYPES.SET_BREADCRUMS,
+    payload: { breadcrumbs },
+  };
+}
+
+/**
+ * Adds a breadcrumb to the end of the list
+ *
+ * @param {object} Breadcrumb to add. Has 'text' and optional 'href' property.
+ * @return {function}
+ */
+export function pushBreadcrumb(breadcrumb) {
```

Comment:
I would prefer if we just allow setting of breadcrumbs, instead of having it behave like a stack. Every view component which wants to set breadcrumbs should know its full context. For example, a GridField listing folder file contents should know all breadcrumbs required for this (parent folders). If you navigate up to the parent folder, *that* GridField should also know all breadcrumbs, rather than knowing about the navigation "direction" (drill down, go back up).

Additionally, "push" and "pop" aren't "safe" state transformations - if you replay them you'll get different results - which should be avoided in a Flux-style state management.